### PR TITLE
Data layers:  Annotations

### PR DESCRIPTION
**Pull request by dprokop**
Adds `AnnotationsDataLayer` to enable annotation queries in scenes.

### CODE REVIEW

File: packages/scenes/src/querying/layers/annotations/AnnotationsDataLayer.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+interface AnnotationsDataLayerState extends SceneObjectState {
+  data?: PanelData;
+  queries: AnnotationQuery[];
+}
+
+export class AnnotationsDataLayer extends SceneObjectBase<AnnotationsDataLayerState> implements SceneDataLayerProvider {
+  private _querySub?: Unsubscribable;
+  private _results = new ReplaySubject<SceneDataLayerProviderResult>();
+
```
Comment (torkelo) [inline packages/scenes/src/querying/layers/annotations/AnnotationsDataLayer.ts:24]: need to add variables dependency here so you can have annotation queries that use variables, but we can add that as a follow-up PR but good to add to a todo list :)

File: packages/scenes/src/querying/layers/annotations/standardAnnotationsSupport.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+  { key: 'prevState' },
+  { key: 'newState' },
+  { key: 'data' as any },
+  { key: 'panelId' },
+  { key: 'alertId' },
+  { key: 'dashboardId' },
+  { key: 'dashboardUID' },
+];
+
+export function getAnnotationsFromData(
```
Comment (torkelo) [inline packages/scenes/src/querying/layers/annotations/standardAnnotationsSupport.ts:150]: do we have any tests for this and other functions here we can copy from core?

Comment (dprokop) [reply packages/scenes/src/querying/layers/annotations/standardAnnotationsSupport.ts:150]: not, still on the todos :)

File: packages/scenes/src/querying/layers/annotations/AnnotationsDataLayer.ts (hunk 1 above)
Comment (dprokop) [reply packages/scenes/src/querying/layers/annotations/AnnotationsDataLayer.ts:24]: yup, added https://github.com/grafana/scenes/issues/313

File: packages/scenes/src/querying/layers/annotations/standardAnnotationsSupport.ts (hunk 2 above)
Comment (dprokop) [reply packages/scenes/src/querying/layers/annotations/standardAnnotationsSupport.ts:150]: done now, moved what I could :)

**Pull request merged**